differ/differ.go

```diff
@@ -301,6 +315,96 @@ func evaluateFilterExpression(eventFilter string, thresholds EventThresholds, ev
 	return evaluator.Evaluate(eventFilter, values)
 }
 
+func findRenderedReport(reports []types.RenderedReport, ruleName types.RuleName, errorKey types.ErrorKey) (types.RenderedReport, error) {
+	for _, report := range reports {
+		reportRuleName := string(report.RuleID)[strings.LastIndex(string(report.RuleID), ".")+1:]
+		if reportRuleName == string(ruleName) && report.ErrorKey == errorKey {
+			return report, nil
+		}
+	}
+	return types.RenderedReport{}, errors.New("report for given rule ID and error key has not been found")
```

Comment:
this could be a constant as well in the // Messages section (and now I think we found all of them)

but most importantly you don't need to import the `errors` package just for this; since there's the package fmt already imported you can use the errorf function, it also offers formatting capabilities and you can print the rule id and error key and have the same exact effect

```
const ReportNotFound = "report for rule ID %v and error key %v has not been found"
[...]
return types.RenderedReport{}, fmt.Errorf(ReportNotFound, ruleName, errorKey )
```

Reply:
Changed